src/rtmp/conn.c

```diff
@@ -788,9 +803,14 @@ int rtmp_connect(struct rtmp_conn **connp, struct dnsc *dnsc, const char *uri,
 	if (!connp || !uri)
 		return EINVAL;
 
-	if (re_regex(uri, strlen(uri), "rtmp://[^:/]+[:]*[0-9]*/[^/]+/[^]+",
-		     &pl_host, NULL, &pl_port, &pl_app, &pl_stream))
+	if (re_regex(uri, strlen(uri), "rtmp://[^/]+/[^/]+/[^]+",
+		     &pl_hostport, &pl_app, &pl_stream)) {
 		return EINVAL;
+	}
+
+	err = decode_hostport(&pl_hostport, &pl_host, &pl_port);
```

Comment:
Suggest we do like this to match existing return code for invalid URIs:
```suggestion
	if (decode_hostport(&pl_hostport, &pl_host, &pl_port))
		return EINVAL;
```